### PR TITLE
Rename Download into Transfer

### PR DESCRIPTION
**Pull request by ezaquarii**
Extending Download into more generic Transfer that
can support both downloads and uploads.

Upload functionality is not provided in this PR - it's
only a rename.

Signed-off-by: Chris Narkiewicz <hello@ezaquarii.com>

### Testing 
- [x] Tests updated

### CODE REVIEW

**Comment by ezaquarii** [comment]
@tobiasKaminsky I'm working on an uploader component and realized that downloader logic is mostly reusable. Instead of providing parallel copy, I'd rather change some concepts to support bi-directional operations.

This PR is an attempt to de-noise the actual Uploader PR from the massive rename change.

There should be no functional changes in this PR.

File: src/main/java/com/nextcloud/client/files/downloader/Direction.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.nextcloud.client.files.downloader
```
Comment (AndyScherzinger) [inline src/main/java/com/nextcloud/client/files/downloader/Direction.kt:1]: lacks a license header

**Comment by AndyScherzinger** [comment]
@ezaquarii looks good to me, only https://github.com/nextcloud/android/pull/6619#discussion_r464319980 Direction should have a license header

File: src/main/java/com/nextcloud/client/files/downloader/Direction.kt (hunk 1 above)
Comment (ezaquarii) [reply src/main/java/com/nextcloud/client/files/downloader/Direction.kt:20]: fixed

**Pull request merged**